### DISCUSSION TITLE
Colors from .Xresources are not used.

### DISCUSSION THREAD

**Discussion by plmi** [bakkeby/st-flexipatch#104] [Q&A] [answered]
I've compiled with
* ALPHA_PATCH
* ALPHA_FOCUS_HIGHLIGHT_PATCH
* BOLD_IS_NOT_BRIGHT_PATCH
* CLIPBOARD_PATCH
* XRESOURCES_PATCH
...and used flexipatch-finalizer on it. I did not touch the resulting config.h.

This is my .Xresources I want to use with st.
```
*background: #1E1E2E
*foreground: #D9E0EE
*.cursorColor: #C3BAC6

! black
*color0: #6E6C7E
*color8: #988BA2

! red
*color1: #F28FAD
*color9: #F28FAD

! green
*color2: #ABE9B3
*color10: #ABE9B3

! yellow
*color3: #FAE3B0
*color11: #FAE3B0

! blue
*color4: #96CDFB
*color12: #96CDFB

! magenta
*color5: #F5C2E7
*color13: #F5C2E7

! cyan
*color6: #89DCEB
*color14: #89DCEB

! white
*color7: #D9E0EE
*color15: #D9E0EE
```

I've executed `xrdb -merge ~/.Xresources`. Some colors are correct but the background color is wrong. It's black instead of `#1E1E2E`. I've noticed this in my config.h:

```
/* bg opacity */
float alpha = 0.8;
float alphaUnfocused = 0.6;

/* Terminal colors (16 first used in escape sequence) */
static const char *colorname[] = {
	/* 8 normal colors */
	"black",
	"red3",
	"green3",
	"yellow3",
	"blue2",
	"magenta3",
	"cyan3",
	"gray90",

	/* 8 bright colors */
	"gray50",
	"red",
	"green",
	"yellow",
	"#5c5cff",
	"magenta",
	"cyan",
	"white",

	[255] = 0,

	/* more colors can be added after 255 to use with DefaultXX */
	"#add8e6", /* 256 -> cursor */
	"#555555", /* 257 -> rev cursor*/
	"#000000", /* 258 -> bg */
	"#e5e5e5", /* 259 -> fg */
};


/*
 * Default colors (colorname index)
 * foreground, background, cursor, reverse cursor
 */
unsigned int defaultbg = 0;
unsigned int bg = 17, bgUnfocused = 16;
unsigned int defaultfg = 259;
unsigned int defaultcs = 256;
unsigned int defaultrcs = 257;
```

`defaultbg` is black which would match my current situation. Is this used instead of `.Xresources`? But other colors work. Is `colorscheme` array used? Do the colors `gray90`, `green` etc. come from `.Xresources`? Why is there `#5c5cff` in the array?

This is the colorschema I've used before using `st-flexipatch`. I want to replicate exactly this colorscheme with `.Xresources`.
```
/* Terminal colors (16 first used in escape sequence) */
static const char *colorname[] = {
	/* 8 normal colors */
	"#6E6C7E",
	"#F28FAD",
	"#ABE9B3",
	"#FAE3B0",
	"#96CDFB",
	"#F5C2E7",
	"#89DCEB",
	"#D9E0EE",

	/* 8 bright colors */
	"#988BA2",
	"#F28FAD",
	"#ABE9B3",
	"#FAE3B0",
	"#96CDFB",
	"#F5C2E7",
	"#89DCEB",
	"#D9E0EE",



[256] = "#D9E0EE", /* default foreground colour */
[257] = "#1E1E2E", /* default background colour */
[258] = "#C3BAC6", /*575268*/

};


/*
 * foreground, background, cursor, reverse cursor
 */
unsigned int defaultfg = 256;
unsigned int defaultbg = 257;
unsigned int defaultcs = 258;
```

**Comment by bakkeby** [comment, marked as answer]
The resource names that are read, and what values they override, are defined in the `resources` array in the st config.

https://github.com/bakkeby/st-flexipatch/blob/1c03f10db9d52c3b9417eeff2e81d0a370280834/config.def.h#L283-L327

A `defaultbg` of 0 just means that it will refer to the `colorname[0]` for the background colour. I'd expect that to end up as `#6E6C7E` rather than black though. Then again you have the alpha patch. I think it may be solid black to have clear transparency, because when you add colours into the mix then the transparency will be tinted. You can change `defaultbg` back to 258 in order for it to read the value that is set by the `background` resource.

The `#5c5cff` in the array comes from default st; added on the 2nd of September, 2010, with a commit comment of "set terminal colors to xterm default ones".

**Reply by plmi** [reply]
Thanks for your quick response. I've solved it by doing this:

```
unsigned int defaultbg = 0;
unsigned int bg = 258, bgUnfocused = 258;
unsigned int defaultfg = 259;
unsigned int defaultcs = 256;
unsigned int defaultrcs = 257;
```

Now focused as well as unfocused terminals have the same color and I can still play with the alpha settings. Thank you very much, you made some really awesome peaces of software!
